dune/grid/common/ZoltanPartition.cpp

```diff
@@ -122,11 +128,62 @@ std::vector<int> zoltanGraphPartitionGridOnRoot(const CpGrid& cpgrid,
         }
 #endif
     }
-    cc.broadcast(&parts[0], parts.size(), root);
+    // free space allocated for zoltan.
     Zoltan_LB_Free_Part(&exportGlobalGids, &exportLocalGids, &exportProcs, &exportToPart);
     Zoltan_LB_Free_Part(&importGlobalGids, &importLocalGids, &importProcs, &importToPart);
     Zoltan_Destroy(&zz);
-    return parts;
+
+    cc.broadcast(&parts[0], parts.size(), root);
+    std::vector<int> my_well_indices;
+
+    if( !pretendEmptyGrid)
```

Comment:
I correct myself. All the processes know the global grid but except one of them all others pretend to not have this information. Thus before load balancing for the load balancer one processes has the whole grid and all the others have an empty partition of the whole grid.